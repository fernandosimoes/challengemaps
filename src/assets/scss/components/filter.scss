%defaultinput {
  border-radius: 5px;
  color: $gray;
  font-size: 20px;
  height: 35px;
  margin-bottom: 15px;
  padding: 0 15px;
  text-align: right;
  width: 320px;

  &::placeholder {
    color: $lightgray;
  }
}

.minimumvalue {
  label {
    font-family: $latobold
  }
}

.minimumvalueinput {
  @extend %defaultinput;
}

.searchinput {
  @extend %defaultinput;
  align-self: center;
  text-align: left;
  width: 100%;
}

.filters {
  justify-content: space-between;
  width: 100%;

  .search {
    align-items: center;
    display: flex;
    width: 100%;
  }
}

.searchinput {
  background: no-repeat 98% center;
  margin-top: 28px;
}

.faturamentominimo {
  align-items: center;
  display: flex;
  font-size: 18px;
  justify-content: space-between;

  label {
    margin-bottom: 15px;
  }

  button {
    margin-top: 10px;
  }

  span {
    color: $red;
    font-size: 12px;
    margin: -15px 0 15px;
  }
}