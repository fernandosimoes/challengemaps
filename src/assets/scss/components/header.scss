.header {
  background: $blue;
  font-size: 35px;
  margin-bottom: 25px;
  width: 100%;

  h1 {
    color: $white;
    font-family: $lato;
    padding: 25px;
  }
}