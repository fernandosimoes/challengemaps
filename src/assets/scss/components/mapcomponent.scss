.mapcontent {
    width: 40%;
}

.minimal--value {
    border-radius: 5px;
    color: $gray;
    height: 35px;
    margin-bottom: 15px;
    text-align: right;
    width: 320px;
    padding: 0 15px;
    font-size: 20px;
    &::placeholder {
        color: $lightgray;
    }
}

.faturamentominimo {
    display: flex;
    flex-direction: column;
    font-size: 25px;

    label {
        margin-bottom: 15px;
    }

    span {
        margin: -15px 0 15px;
        font-size: 12px;
        color: $red;
    }
}