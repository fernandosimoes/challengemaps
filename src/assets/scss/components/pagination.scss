.pagination {

  button {
    background: none;
    border: none;
    cursor: pointer;

    &.selected {
      background-color: $lightgray;

    }
  }
}
