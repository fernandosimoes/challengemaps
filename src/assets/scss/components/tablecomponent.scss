.tablecontent {
    width: 100%;
}

.table {
    border-bottom: 0;
    border-radius: 5px;
    border: 1px solid $gray;

    .line {
        border-bottom: 1px solid $gray;
        display: flex;
        justify-content: space-between;
        width: 100%;
        span {
            font-size: 18px;
            margin:0 5px;
            padding: 5px 0;
        }

        a {
            font-size: 18px;
            margin:0 5px;
            padding: 5px 0;
            text-decoration: none;
            color: $gray;
            font-weight: bold;
        }

        .low {
            color: $red
        }
    }
}