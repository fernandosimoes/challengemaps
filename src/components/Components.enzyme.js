// Link.react.test.js
import React from 'react';
import Header from './Header';
import { MapContainer } from './Map';

import { configure, shallow, mount, render } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

test('Enzyme Map', () => {
    // Render a checkbox with label in the document
    const map = shallow(<MapContainer />);
    // expect(header.hasClass('header')).toBe(true);
    // expect(wrapper.find('.my-button').hasClass('disabled')).to.equal(true);
    // expect(checkbox.text()).toEqual('Off');

    // checkbox.find('input').simulate('change');

    // expect(checkbox.text()).toEqual('On');
});