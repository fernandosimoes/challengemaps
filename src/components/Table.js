import React, { Component } from 'react';

class Table extends Component {
    render() {
        return (
            <div className="tablecontent">
                
            </div>
        );
    }
}

export default Table;