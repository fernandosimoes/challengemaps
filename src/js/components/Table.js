import React, { Component } from 'react';
import Pagination from "./Pagination";

import { changepage, sorteValue, sorteByName } from "../actions/lojasaction";
import { connect } from 'react-redux';
import CurrencyFormat from 'react-currency-format';

class TableComponent extends Component {

    constructor(props) {
        super(props);
        this.changelocalpage = this.changelocalpage.bind(this)
        this.orderByValue = this.orderByValue.bind(this)
        this.orderByName = this.orderByName.bind(this)
        this.state = {
            direction: 'asc',
            directionName: 'asc'
        }
    }

    changelocalpage(page) {
        if (page < 0 || page >= this.props.splitedPages.length) {
            return false;
        }
        this.props.changepage(page);

    }
    orderByName() {

        if (this.state.directionName == 'asc') {
            this.setState({ directionName: 'desc' })
        } else {
            this.setState({ directionName: 'asc' })
        }
        this.props.sorteByName(this.state.directionName, this.props.todaslojas)
    }
    orderByValue(direction) {
        let sortedValue;
        if(this.state.direction == 'asc') {
            sortedValue = this.props.lojas.sort((a, b) => {
                return a.revenue - b.revenue;
            })
            this.setState({direction: 'desc'})
        } else {
            sortedValue = this.props.lojas.sort((a, b) => {
                return b.revenue - a.revenue;
            })
            this.setState({ direction: 'asc' })
        }
        this.props.sorteValue(sortedValue)
    }

    render() {

        if (!this.props.splitedPages.length){
            return "loading";
        }
        return (
            <div className="tablecontent">
                <div className="table">
                    <p className="line">
                        <a className="lojaname" onClick={this.orderByName}>
                            Loja
                        </a>
                        <a className="faturamento" onClick={this.orderByValue}>
                            Faturamento
                        </a>
                    </p>
                    {this.props.splitedPages[this.props.currentPage].map((store, kstore)=>{
                        if (store.revenue < this.props.minimumvalue) {
                            return (<p key={kstore} className="line">
                                <span className="lojaname">{store.name}</span>
                                <span className="faturamento low">{<CurrencyFormat value={store.revenue} displayType={'text'} thousandSeparator={true} prefix={'R$ '} />}</span>
                            </p>)
                        } else {
                            return (<p key={kstore} className="line">
                                <span className="lojaname">{store.name}</span>
                                <span className="faturamento">{<CurrencyFormat value={store.revenue} displayType={'text'} thousandSeparator={true} prefix={'R$ '} />}</span>
                            </p>)
                        }

                    })}
                </div>
                <Pagination changepage={this.changelocalpage} totalPages={this.props.splitedPages.length} currentPage={this.props.currentPage}/>
            </div>
        );
    }
    componentDidMount() {
    }
}


const mapStateToProps = (state, ownProps) => {
    return {
        lojas: state.lojas,
        todaslojas: state.todaslojas,
        currentPage: state.currentPage,
        splitedPages: state.splitedPages,
        minimumvalue: state.minimumvalue
    }
}


const mapDispatchToProps = (dispatch, ownProps) => {
    return {
        changepage: (page) => { dispatch(changepage(page)) },
        sorteValue: (ordered) => { dispatch(sorteValue(ordered)) },
        sorteByName: (direction, lojas) => { dispatch(sorteByName(direction, lojas)) }
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(TableComponent)
// export default Filters;
