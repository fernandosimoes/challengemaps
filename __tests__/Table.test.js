// Link.react.test.js
import React from 'react';
import { TableComponent } from '../src/js/components/Table';
import renderer from 'react-test-renderer';
import {configure, shallow, mount} from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import {allstores, splitedPages} from './fakedata';

describe('Table Component', () => {
    // Render a checkbox with label in the document
    const table = mount(<TableComponent
        lojas={allstores.stores}
        todaslojas={allstores.stores}
        currentPage={0}
        splitedPages={splitedPages}
        minimumvalue={15000}
        sorteByName={()=>{return false}} // this function return false because she dispatch changes in the redux then in the test, she dont do this dispach
        sorteValue={()=>{return false}} // this function return false because she dispatch changes in the redux then in the test, she dont do this dispach
        changepage={(page)=>{ console.log(page) }} // this function return false because she dispatch changes in the redux then in the test, she dont do this dispach
    />);

    test('Render Component', () => {

        let component = renderer.create(
            <TableComponent
                lojas={allstores.stores}
                todaslojas={allstores.stores}
                currentPage={0}
                splitedPages={splitedPages}
                minimumvalue={15000}
            />
        )
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('Dom Clicks Sort By Name',() =>{
        table.find('a.lojaname').simulate('click');
        expect(table.state().directionName).toEqual('desc');
    })

    test('Dom Clicks Sort By Value',() =>{
        table.find('a.faturamento').simulate('click');
        expect(table.state().direction).toEqual('desc');
    })
})