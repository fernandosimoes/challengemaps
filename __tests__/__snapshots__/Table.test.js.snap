// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Component Render Component 1`] = `
<div
  className="tablecontent"
>
  <div
    className="table"
  >
    <p
      className="line"
    >
      <a
        className="lojaname"
        onClick={[Function]}
      >
        Loja
      </a>
      <a
        className="faturamento"
        onClick={[Function]}
      >
        Faturamento
      </a>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Alameda Santos
      </span>
      <span
        className="faturamento"
      >
        <span>
          R$ 29,854.12
        </span>
      </span>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Av Paulista
      </span>
      <span
        className="faturamento"
      >
        <span>
          R$ 31,257.23
        </span>
      </span>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Av Paulista Novo
      </span>
      <span
        className="faturamento"
      >
        <span>
          R$ 19,874.25
        </span>
      </span>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Bandeirantes
      </span>
      <span
        className="faturamento"
      >
        <span>
          R$ 21,587.33
        </span>
      </span>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Bar√£o do Triunfo
      </span>
      <span
        className="faturamento low"
      >
        <span>
          R$ 12,569.95
        </span>
      </span>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Bela Vista
      </span>
      <span
        className="faturamento"
      >
        <span>
          R$ 22,365.54
        </span>
      </span>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Brigadeiro Faria Lima
      </span>
      <span
        className="faturamento"
      >
        <span>
          R$ 17,988.36
        </span>
      </span>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Brooklin
      </span>
      <span
        className="faturamento low"
      >
        <span>
          R$ 12,578.22
        </span>
      </span>
    </p>
    <p
      className="line"
    >
      <span
        className="lojaname"
      >
        Brooklin 2
      </span>
      <span
        className="faturamento low"
      >
        <span>
          R$ 12,589.27
        </span>
      </span>
    </p>
  </div>
  <div
    className="pagination"
  >
    <button
      className="back"
      onClick={[Function]}
    >
      <img
        src=""
      />
    </button>
    <button
      className="pagenumber selected"
      onClick={[Function]}
    >
      1
    </button>
    <button
      className="pagenumber"
      onClick={[Function]}
    >
      2
    </button>
    <button
      className="pagenumber"
      onClick={[Function]}
    >
      3
    </button>
    <button
      className="next"
      onClick={[Function]}
    >
      <img
        src=""
      />
    </button>
  </div>
</div>
`;
